src/lib/comms/sol-coap.c

```diff
@@ -634,50 +634,22 @@ pending_reply_free(struct pending_reply *reply)
     free(reply);
 }
 
-static bool
-user_wants_to_retry_by_id(struct sol_coap_server *server, uint16_t id)
-{
-    uint16_t i;
-    struct pending_reply *reply;
-
-    SOL_PTR_VECTOR_FOREACH_REVERSE_IDX (&server->pending, reply, i) {
-        if (reply->id != id)
-            continue;
-
-        if (reply->cb(server, NULL, NULL, (void *)reply->data))
-            return true;
-        sol_ptr_vector_del(&server->pending, i);
-        pending_reply_free(reply);
-    }
-
-    return false;
-}
-
-static bool
-user_wants_to_retry(struct sol_coap_server *server, struct sol_coap_packet *pkt)
-{
-    uint16_t id;
-
-    sol_coap_header_get_id(pkt, &id);
-
-    return user_wants_to_retry_by_id(server, id);
-}
-
 /* This is mostly for !CON packets, which do not go to the outgoing
  * list but also keep a context of their own, for response handling */
 static bool
 pending_timeout_cb(void *data)
 {
-    struct pending_reply *pending_reply = data;
-    struct sol_coap_server *server = pending_reply->server;
+    struct pending_reply *reply = data;
+    struct sol_coap_server *server = reply->server;
 
-    /* The pending context will be kept, as the user wishes */
-    if (user_wants_to_retry_by_id(server, pending_reply->id))
-        return true;
+    reply->cb(server, NULL, NULL, (void *)reply->data);
```

Comment:
I was respecting our documentation, which had a (for me, fair) contract about this function's return value:

```
 * After an internal timeout is reached, @a reply_cb will be called
 * with @c NULL @c req and @c cliaddr. The same behavior is expected
 * for its return: if @c true, @a server will issue a new timeout and
 * continue waiting responses until it ends, otherwise @a server will
 * terminate response waiting.
```